src/main/webapp/main.js

```diff
@@ -30,3 +32,47 @@ Promise.all([mapPromise, domPromise]).then(() => {
 function initMap() {
   myMap = new ChapMap();
 }
+
+
+connection.onopen = () => {
+};
+
+connection.onclose = () => {
+};
+
+connection.onerror = (error) => {
+  console.error(error);
+};
+
+connection.onmessage = (event) => {
+  let li = document.createElement('li');
+  
+  var obj = JSON.parse(event.data); 
+
+  if(obj.type !== 'MSG_RECV'){
+      // let Alice deal with it;
+      // obj is a JSON object
+  } else {
+      li.innerText = obj.uid + ": " + obj.message;
+      document.querySelector('#chat').append(li);
+  }  
+};
+
+function getServerUrl() {
+    
+    var defaultChatRoomID = "1234goroom";
+    var protoSpec;
+    var defaultIDToken = 12;
+
+    if (location.protocol !== 'https:' && location.hostname != 'localhost:8080') {
+        location.replace(`https:${location.href.substring(location.protocol.length)}`);
+    } 
+
+    if(location.protocol === 'https:'){
+        protoSpec = 'wss:';
+    } else {
+        protoSpec = 'ws:';
+    }
+
+    return location.protocol + "//" + location.hostname + "/chat/" + defaultChatRoomID + "?idToken=" + idToken;
```

Comment:
Apply the change from `location.protocol` to `protoSpec` in the return statement.

Reply:
Gotcha! Thanks for pointing this out!